File: 238 Product of Array Except Self/Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = [1] * len(nums)
        prefix =1 
        postfix = 1
        for a in range(len(nums)):
            result[a] = prefix
            prefix *= nums[a]
        for b in range(len(nums)-1,-1,-1):
             result[b] *= postfix
             postfix *= nums[b]
        return result
 
        

File: 238 Product of Array Except Self/Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int length = nums.size();
        vector<int> result(length, 1);

        int prefix = 1;
        for(int i = 0; i < length; i++) {
            result[i] *= prefix;
            prefix *= nums[i];
        }

        int postfix = 1;
        for(int i = length - 1; i >= 0; i--) {
            result[i] *= postfix;
            postfix *= nums[i];
        }

        return result;
    }
};
